src/java.desktop/windows/classes/sun/awt/windows/ThemeReader.java

```diff
@@ -107,14 +107,23 @@ private static Long openThemeImpl(String widget, int dpi) {
            // We're using the syntax "subAppName::controlName" here, as used by msstyles.
            // See documentation for SetWindowTheme on MSDN.
            setWindowTheme(widget.substring(0, i));
-           theme = openTheme(widget.substring(i + 2), dpi);
+           theme = getOpenThemeValue(widget.substring(i + 2), dpi);
```

Comment:
I suggest changing the type of `theme` variable from `Long` to `long` as well as the return value of `openThemeImpl`.

Reply:
Means only till openThemeImpl return value and not further up the hierarchy ?

Reply:
Yes, only the low-level methods until it's put into a map.

Reply:
Yeah, sounds good. Will do that.